apps/freeablo/faworld/monster.h

```diff
@@ -10,18 +11,30 @@ namespace DiabloExe
 
 namespace FAWorld
 {
+    class AI;
+    class NullAI;
+
     class Monster: public Actor
     {
         STATIC_HANDLE_NET_OBJECT_IN_CLASS()
 
         public:
             Monster();
             Monster(const DiabloExe::Monster& monster, Position pos);
+            ~Monster();
             void init();
             std::string getDieWav();
             std::string getHitWav();
+            void attachAI(AI* ai) {
+              mAi = ai;
+            };
+
+            bool attack(Actor * enemy);
+            void update(bool noclip, size_t ticksPassed);
+            bool canIAttack(Actor * actor);
 
         private:
+            AI * mAi;
```

Comment:
I would implement this on Actor, then maybe having a special "playerAI" that allows an actor to be controlled by keyboard and mouse IE, the player character. In this case, it might be better to rename AI to behaviour, but that might be overkill. I'd definitely move the code to Actor though, anyway